internal/services/applications/parse/from_template.go

```diff
@@ -95,3 +95,21 @@ func (id FromTemplateId) Segments() []resourceids.Segment {
 func (id FromTemplateId) String() string {
 	return fmt.Sprintf("Application From Template (Template ID: %q, Application ID: %q, Service Principal ID: %q)", id.TemplateId, id.ApplicationId, id.ServicePrincipalId)
 }
+
+func (id FromTemplateId) FromParseResult(input resourceids.ParseResult) error {
```

Comment:
This'll need to be a pointer:

```suggestion
func (id *FromTemplateId) FromParseResult(input resourceids.ParseResult) error {
```